spec/features/add_an_ip_address_spec.rb

```diff
@@ -9,61 +9,64 @@
   include_examples 'confirmation use case spy'
   include_examples 'notifications service'
 
-  context 'when logged out' do
+  context 'when logged in' do
     before do
+      sign_in_user create(:user)
       visit new_ip_path
     end
 
-    it_behaves_like 'not signed in'
-  end
+    it_behaves_like 'shows activation notice'
 
-  context 'when logged in' do
-    let(:user) { create(:user, :confirmed, :with_organisation) }
-    before do
-      sign_in_user user
-      visit ips_path
-      expect(page).to have_content "Add IP"
-      click_on "Add IP Address"
+    it 'asks me to enter an IP' do
+      expect(page).to have_content('Enter IP Address (IPv4 only)')
     end
 
-    context 'before saving the IP' do
-      it_behaves_like 'shows activation notice'
+    context 'and that IP is valid' do
+      before do
+        fill_in 'address', with: '10.0.0.1'
+        click_on 'Save'
+      end
+
+      it 'shows a success message' do
+        expect(page).to have_content('IP Added')
+      end
 
-      it 'displays the form' do
-        expect(page).to have_content('Enter IP Address (IPv4 only)')
+      it 'shows me a list of IPs including the new IP' do
+        expect(page).to have_content('10.0.0.1')
       end
     end
 
-    context 'with an invalid IP address' do
+    context 'and that IP is invalid' do
       before do
         fill_in 'address', with: 'InvalidIP'
-        expect { click_on 'Save' }.to change { Ip.count }.by(0)
+        click_on 'Save'
       end
 
       it_behaves_like 'errors in form'
 
-      it 'displays the form with error message' do
+      it 'asks me to re-enter my IP' do
         expect(page).to have_content('Enter IP Address')
+      end
+
+      it 'tells me what I entered was invalid' do
         expect(page).to have_content(
           'Address must be a valid IPv4 address (without subnet)'
         )
       end
-    end
-
-    context 'after successfully saving an IP' do
-      before do
-        fill_in 'address', with: '10.0.0.1'
-        expect { click_on 'Save' }.to change { Ip.count }.by 1
-      end
 
-      it_behaves_like 'shows activation notice'
+      context 'when looking back at the list' do
+        before { visit ips_path }
 
-      it 'shows the successful confirmation page' do
-        expect(Ip.first.user).to eq(user)
-        expect(page).to have_content('IP Added')
-        expect(page).to have_content(Ip.first.address)
-        expect(page).to have_content(user.radius_secret_key)
+        it 'has not added the invalid IP' do
+          expect(page).to_not have_content('InvalidIP')
```

Comment:
Is "InvalidIP" an error that we display to the users?

Reply:
No, it's the string we entered in the previous step as an example of an invalid IP. 

Maybe some junk data makes this clearer (e.g. "121.141.121.aaa")?

Reply:
Changed this in [this commit](https://github.com/alphagov/govwifi-admin/pull/83/commits/ded50e6ad8e765d89fb5601b7f55e15718564723).